paddle/fluid/framework/ir/memory_optimize_pass/memory_reuse_pass.cc

```diff
@@ -41,6 +41,15 @@ void MemoryReusePass::ApplyImpl(Graph *graph) const {
     pinned_var_set_ = &graph->Get<details::PinnedVars>(details::kPinnedVars);
   }
 
+  skip_reuse_vars_ = &Get<std::vector<SkipReuseVars>>(kSkipReuseVars);
+  if (skip_reuse_vars_->empty()) {
+    skip_reuse_vars_->resize(all_vars_->size());
+  } else {
+    PADDLE_ENFORCE_EQ(
+        skip_reuse_vars_->size(), all_vars_->size(),
+        platform::errors::InvalidArgument("Size not match, this may be a bug"));
```

Comment:
Why skip_reuse_vars_.size() should equal all_vars_->size() or empty?

Reply:
Thanks a lot! I have added some comments in codes. `all_vars_->size()` returns the place number inside `ParallelExecutor`, so `skip_reuse_vars_` must be not initialized (empty) or have the same place number as `all_vars_`.